solidity/test/VendingMachineTest.js

```diff
@@ -531,6 +531,29 @@ describe("VendingMachine", async function() {
           "Bad _extraData signature. Call must be to tbtcToBtc.",
         )
       })
+
+      it("protects receiveApproval from external callers", async () => {
+        const tbtcToBtc = vendingMachine.abi.filter(
+          x => x.name == "tbtcToBtc",
+        )[0]
+        const calldata = web3.eth.abi.encodeFunctionCall(tbtcToBtc, [
+          testDeposit.address,
+          "0x1111111100000000",
+          redeemerOutputScript,
+          owner,
+        ])
+
+        await expectRevert(
+          redemptionScript.receiveApproval(
+            redemptionScript.address,
+            tdtId,
+            owner,
+            calldata,
```

Comment:
Feels like this can just be an empty byte array without losing the test, right?

Reply:
yep